packages/web3/src/utils/exchange.ts

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import { AddressType, DUST_AMOUNT, addressFromPublicKey, addressFromScript, binToHex, bs58, hexToBinUnsafe } from '..'
+import { Transaction } from '../api/api-alephium'
+import { Address } from '../signer'
+
+enum UnlockScriptType {
+  P2PKH = 0x00,
+  P2MPKH = 0x01,
+  P2SH = 0x02
+}
+
+export function isDepositALPHTransaction(tx: Transaction, exchangeAddress: string): boolean {
+  return isDepositTransaction(tx, exchangeAddress) && checkALPHOutput(tx)
+}
+
+export function isDepositTokenTransaction(tx: Transaction, exchangeAddress: string): boolean {
+  return isDepositTransaction(tx, exchangeAddress) && checkTokenOutput(tx, exchangeAddress)
+}
+
+// we assume that the tx is deposit transaction
+export function getDepositAddress(tx: Transaction): Address {
+  return getAddressFromUnlockScript(tx.unsigned.inputs[0].unlockScript)
+}
+
+export function isExchangeAddress(address: string): boolean {
+  const decoded = bs58.decode(address)
+  if (decoded.length === 0) throw new Error('Address is empty')
+  const addressType = decoded[0]
+  return addressType === AddressType.P2PKH || addressType === AddressType.P2SH
+}
+
+export function getAddressFromUnlockScript(unlockScript: string): Address {
```

Comment:
Do we want to expose this?

Reply:
We need to expose it because we are using it in unit tests, but now in `index.ts` it is not exposed.